src/server/pmix_server_get.c

```diff
@@ -631,7 +638,7 @@ static pmix_status_t _satisfy_request(pmix_nspace_t *nptr, pmix_rank_t rank,
         cb.proc = &proc;
         cb.scope = scope;
         cb.copy = false;
-        PMIX_GDS_FETCH_KV(rc, peer, &cb);
+        PMIX_GDS_FETCH_KV(rc, pmix_globals.mypeer, &cb);
```

Comment:
Something about this doesn't feel quite right. In the above code, we store the incoming data in the GDS module for a peer in this nspace if the scope was LOCAL, and in ```pmix_globals.mypeer``` otherwise - we point the ```peer``` object at the GDS module where we store the data. So why then are we looking for the data strictly in ```pmix_globals.mypeer```? Won't this only be correct for scope != LOCAL?